WalletWasabi.Backend/Controllers/BlockchainController.cs

```diff
@@ -416,4 +420,49 @@ private async Task<StatusResponse> FetchStatusAsync(CancellationToken cancellati
 
 		return status;
 	}
+
+	[HttpGet("get-transaction-fee")]
+	[ProducesResponseType(200)]
+	[ProducesResponseType(400)]
+	public async Task<IActionResult> GetTransactionFeeAsync([FromQuery, Required] string transactionId, CancellationToken cancellationToken)
+	{
+		uint256 txID = new(transactionId);
+
+		lock (TransactionFeeCacheLock)
+		{
+			if (TransactionFeeCache.TryGetValue(txID, out Money? value))
+			{
+				return Ok(value);
+			}
+		}
+
+		List<Coin> inputs = new();
+		Dictionary<uint256, Transaction> parentTransactions = new();
+
+		var tx = await RpcClient.GetRawTransactionAsync(txID, true, cancellationToken);
+
+		foreach (var input in tx.Inputs)
+		{
+			if (!parentTransactions.ContainsKey(input.PrevOut.Hash))
+			{
+				var parentTx = await RpcClient.GetRawTransactionAsync(input.PrevOut.Hash, true, cancellationToken).ConfigureAwait(false);
+				parentTransactions.Add(input.PrevOut.Hash, parentTx);
+			}
+
+			TxOut txOut = parentTransactions[input.PrevOut.Hash].Outputs[input.PrevOut.N];
+			inputs.Add(new Coin(input.PrevOut, txOut));
+		}
+
+		Money fee = tx.GetFee(inputs.ToArray());
+
+		lock (TransactionFeeCacheLock)
+		{
+			if (TransactionFeeCache.TryAdd(txID, fee) && TransactionFeeCache.Count > 1000)
+			{
+				TransactionFeeCache.Remove(TransactionFeeCache.Keys.First());
+			}
```

Comment:
Naive invalidating method, so the cache doesn't grow indefinitely.

Reply:
Yes, it would make sense, but can I leave this suggestion for a follow-up PR?

Reply:
As mentioned IDK where, yes